Lib/ntpath.py

```diff
@@ -135,7 +135,7 @@ def join(path, *paths):
             result_path = result_path + p_path
         ## add separator between UNC and non-absolute path
         if (result_path and not result_root and
-            result_drive and result_drive[-1:] not in colon + seps):
+            result_drive and result_drive[-1] not in colon + seps):
```

Comment:
This change means that IndexError will be raised if result_drive is an empty string. Does the change speed anything up?

Reply:
The `result_drive and ` bit at the start of the line already ensures that `result_drive` is non-empty. The real reason for the slice is that directly indexing `result_drive` will give you an `int` if the function is called with a `bytes` input.

Reply:
Wasn't handled correctly a few lines higher.

Reply:
Oh right, but indexing into `bytes` using an `int` works fine:

```
>>> 47 in b'/'
True
```

If I knew about this I must have forgotten.

Reply:
OK, I'll remove the slice.